internal/provider/data_source_certificate_test.go

```diff
@@ -639,3 +639,45 @@ func TestDataSourceCertificate_MalformedURL(t *testing.T) {
 		},
 	})
 }
+
+func TestDataSourceCertificate_UnknownComputedCertificatesUntilApplied(t *testing.T) {
+	r.Test(t, r.TestCase{
+		ProtoV5ProviderFactories: protoV5ProviderFactories(),
+
+		Steps: []r.TestStep{
+			{
+
+				Config: `
+					resource "tls_private_key" "test" {
+						algorithm = "ED25519"
+					}
+
+					data "tls_certificate" "test" {
+						url = replace(tls_private_key.test.id, "/^.*$/","https://terraform.io")
```

Comment:
Nit: May be good to leave a code comment here explaining what it happening for future readers

```suggestion
						# This attribute value must be unknown to trigger
						# the behavior, therefore this replaces the unknown
						# value with a known value on apply, so the URL is valid.
						url = replace(tls_private_key.test.id, "/^.*$/","https://terraform.io")
```